dartagnan/src/main/java/com/dat3m/dartagnan/Dartagnan.java

```diff
@@ -75,13 +76,30 @@ public class Dartagnan extends BaseOptions {
 
     private static final Logger logger = LogManager.getLogger(Dartagnan.class);
 
-    private static final Set<String> supportedFormats =
-            ImmutableSet.copyOf(Arrays.asList(".litmus", ".c", ".i", ".ll", ".spv.dis"));
+    private static final Set<String> supportedFormats = ImmutableSet.copyOf(ProgramParser.SUPPORTED_EXTENSIONS);
 
     private Dartagnan(Configuration config) throws InvalidConfigurationException {
         config.recursiveInject(this);
     }
 
+    private static Configuration loadConfiguration(String[] args) throws InvalidConfigurationException, IOException {
+        ConfigurationBuilder builder = Configuration.builder();
+        for (String argument : args) {
+            int index = argument.indexOf('=');
+            if (index != -1 && argument.startsWith("--")) {
+                builder.setOption(argument.substring(2, index), argument.substring(index + 1));
+            } else if (argument.endsWith(".properties")) {
+                try {
+                    builder.loadFromFile(argument);
+                    logger.info("Loaded properties from {}", argument);
+                } catch (IOException e) {
+                    logger.warn("Could not load file {}: {}", argument, e.getMessage());
+                }
+            }
+        }
+        return builder.build();
+    }
+
```

Comment:
I feel like this is reinventing the wheel :)
Your methods is like a mixture of `Configuration.fromCmdLineArguments` and `Configuration.builder().loadFromSource`, the latter of which supports `#include` directives.
I have the feeling that one of the following is possible:
- Generate a single `CharSource` from the arguments and use `loadFromSource`
- Split the args into loads of `.properties` files (not sure what syntax you want to use) and standard arguments. Then use `fromCmdLineArguments` on the former and `loadFromSource/File` on the latter (`loadFromSource` would also support nested loadings with `#include`)

Reply:
The facility of `loadFromSource` looks very useful.  Sadly, `fromCmdLineArguments` is not a method of `ConfigurationBuilder`, but with `CharSource`, there is little left to code.

Reply:
While `fromCmdLineArguments` is not a builder method, you can still generate a full `Configuration` and then create a builder starting from that config via `ConfigurationBuilder.copyFrom(Configuration)`. With that builder, you can then import whatever extra files you want.
Though I guess it would be better to first load the file and then the remaining args, so that they overwrite whatever the file says. I mean, if the user explictily specifies something that is also in a file, then I guess the wants to overwrite it.

Reply:
I got an `IllegalStateException` from `copyFrom`, because it is not the first operation on a builder.  Instead, I added the file arguments as `#include` directives on top of the `CharSource`.